packages/admin/src/src/dialogs/AddInstanceDialog.tsx

```diff
@@ -0,0 +1,329 @@
+    },
+    messageTitle_info: {
+        background: '#5abd29',
+        borderRadius: 3,
+        paddingLeft: 10,
+        fontWeight: 'bold',
+        color: theme.palette.mode === 'dark' ? 'black' : 'white',
+    },
+    deps: {
+        margin: 10,
+        fontSize: 16,
+        color: theme.palette.mode === 'dark' ? '#e70000' : '#840101',
+    },
+});
+
+interface AdapterDependencies {
+    name: string;
+    version: string | null;
+    installed: string;
+    installedVersion: string;
+    rightVersion: boolean;
+}
+
+interface AddInstanceDialogProps {
+    adapter: string;
+    instancesWorker: InstancesWorker;
+    socket: AdminConnection;
+    currentHost: string;
+    currentInstance: string;
+    t: (text: string, arg1?: any, arg?: any) => string;
+    onClick: () => void;
+    onClose: () => void;
+    onHostChange: (host: string) => void;
+    onInstanceChange: (event: SelectChangeEvent<string>) => void;
+    instances: Record<string, any>;
+    adapterObject: ioBroker.AdapterCommon;
+    classes: Record<string, any>;
+    dependencies: AdapterDependencies[];
+    hostsWorker: InstanceType<typeof HostsWorker>;
+    noTranslation: boolean;
+    toggleTranslation: () => void;
+    expertMode: boolean;
+}
+
+interface AddInstanceDialogState {
+    instanceNumbers: string[];
+}
+
+class AddInstanceDialog extends Component<AddInstanceDialogProps, AddInstanceDialogState> {
+    private readonly t: (text: string, arg1?: any, arg?: any) => string;
+
+    private readonly messages: Message[] | null = null;
+
+    private readonly lang: string;
+
+    constructor(props: AddInstanceDialogProps) {
+        super(props);
+        this.state = {
+            instanceNumbers: [],
+        };
+        this.t = props.t;
+        this.lang = I18n.getLanguage();
+
+        this.messages = checkCondition(
+            (props.adapterObject as any).messages as Rule[], // TODO: remove any
+            null,
+            props.adapterObject.version,
+            props.instances,
+        );
+    }
+
+    componentDidMount() {
+        this.props.instancesWorker.getInstances()
+            .then((instances: Record<string, ioBroker.InstanceObject>) => {
+                const instanceNumbers = Object.keys(instances)
+                    .filter(id => instances[id]?.common?.name === this.props.adapter)
+                    .map(id => id.substring(id.lastIndexOf('.') + 1));
+
+                this.setState({ instanceNumbers });
+            });
+    }
+
+    getAvailableInstances() {
+        const result = [];
+        result.push(<MenuItem value="auto" key="auto">{this.t('auto')}</MenuItem>);
+
+        for (let i = 0; i <= 10; i++) {
+            if (!this.state.instanceNumbers.includes(i.toString())) {
+                result.push(<MenuItem value={i} key={i}>{i}</MenuItem>);
+            }
+        }
+
+        return result;
+    }
+
+    checkDependencies = (dependencies = this.props.dependencies) => {
```

Comment:
any reason for the arrow function assigment here instead of normal class method?